src/services/stringCompletions.ts

```diff
@@ -1072,9 +1153,11 @@ function getModulesForPathsPattern(
 
     const normalizedSuffix = normalizePath(parsed.suffix);
     const declarationExtension = normalizedSuffix && getDeclarationEmitExtensionForPath("_" + normalizedSuffix);
-    const matchingSuffixes = declarationExtension ? [changeExtension(normalizedSuffix, declarationExtension), normalizedSuffix] : [normalizedSuffix];
+    const inputExtension = isImports && normalizedSuffix ? getPossibleOriginalInputExtensionForExtension("_" + normalizedSuffix) : undefined;

```

Comment:
All of the logic in this function assumes the same extensions between potential inputs and outputs and stuff. This can impact the used globs but I think it's largely fine right now - unless somebody would come up with a compelling test case that should work differently.